### PR TITLE
WIN 48

### PR DESCRIPTION
**Pull request by AbhijitSinha1**
Add validation engine, and utility methods required for conversion from `uint` to `bytes32` or `uint[]`  to `bytes32[]`

### CODE REVIEW

File: validation/contracts/ValidationEngine.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    ruleSets[profileName][ruleName][false] = ruleSets[profileName][ruleName][true];
+    ruleSets[profileName][ruleName][true]  = ValidationRuleInterface(0);
+  }
+
+  function activateRule(bytes32 profileName, bytes32 ruleName) public {
+    ruleSets[profileName][ruleName][true]  = ruleSets[profileName][ruleName][false];
+    ruleSets[profileName][ruleName][false] = ValidationRuleInterface(0);
+  }
+
+  function validate(address contractAddress, bytes32 profileName) public returns (bool, uint, bytes32) {
```
Comment (liorsaar1) [inline validation/contracts/ValidationEngine.sol]: please change to rest status:
`return (RestStatus.BAD_REQUEST, WingsError.PROFILE_NOT_FOUND, null);`
or
`return (RestStatus.OK, WingsError.NULL, payload);`

File: validation/contracts/ValidationStatus.sol [hunk 2]
```diff
@@ -0,0 +1,4 @@
+contract ValidationStatus {
```
Comment (liorsaar1) [inline validation/contracts/ValidationStatus.sol]: please use rest codes only in the library.  this belongs in the app

**Comment by samrit** [comment]
The return type needs to be of the following type
`return (uint, uint, bool, bytes32)` 
* The first `uint` is the `RestStatus`. This should be `OK` in all cases except when the profile is not found, in which case it is `NOT_FOUND`
* The second `uint` is the validation error code, specific to the application.
* The remaining params are any additional responses you might need, which in this case is a `bool` and `bytes32`
We are standardizing return types around this structure

File: validation/contracts/ValidationStatus.sol (hunk 2 above)
Comment (samrit) [reply validation/contracts/ValidationStatus.sol]: This needs to be present, and will be extended by the app.

**Pull request merged**